jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/HibernateSearchProcessor.java

```diff
@@ -77,42 +83,67 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
-                    else if (!providerModule.equals(searchModuleName)) {
-                        throw JpaLogger.ROOT_LOGGER.differentSearchModuleDependencies(deploymentUnit.getName(), searchModuleName, providerModule);
+                    else if (!providerModule.equals(searchMapperModuleName)) {
+                        throw JpaLogger.ROOT_LOGGER.differentSearchModuleDependencies(deploymentUnit.getName(), searchMapperModuleName, providerModule);
                     }
                 }
             }
         }
 
-        if (NONE.equals(searchModuleName)) {
+        if (NONE.equals(searchMapperModuleName)) {
             // Hibernate Search module will not be added to deployment
             ROOT_LOGGER.debugf("Not adding Hibernate Search dependency to deployment %s", deploymentUnit.getName());
             return;
         }
 
         // use Search module name specified in persistence unit definition
-        if (searchModuleName != null && !IGNORE.equals(searchModuleName)) {
-            ModuleIdentifier moduleIdentifier = ModuleIdentifier.fromString(searchModuleName);
+        if (searchMapperModuleName != null && !IGNORE.equals(searchMapperModuleName)) {
+            ModuleIdentifier moduleIdentifier = ModuleIdentifier.fromString(searchMapperModuleName);
             moduleSpecification.addSystemDependency(new ModuleDependency(moduleLoader, moduleIdentifier, false, true, true, false));
             ROOT_LOGGER.debugf("added %s dependency to %s", moduleIdentifier, deploymentUnit.getName());
         } else {
             // add Hibernate Search module dependency if application is using the Hibernate Search Indexed annotation
             final CompositeIndex index = deploymentUnit.getAttachment(org.jboss.as.server.deployment.Attachments.COMPOSITE_ANNOTATION_INDEX);
-            List<AnnotationInstance> annotations = index.getAnnotations(SEARCH_INDEXED_ANNOTATION_NAME);
+            List<AnnotationInstance> annotations = index.getAnnotations(ANNOTATION_INDEXED_NAME);
             if (annotations != null && !annotations.isEmpty()) {
-                moduleSpecification.addSystemDependency(new ModuleDependency(moduleLoader, defaultSearchModule, false, true, true, false));
-                ROOT_LOGGER.debugf("deployment %s contains %s annotation, added %s dependency", deploymentUnit.getName(), SEARCH_INDEXED_ANNOTATION_NAME, defaultSearchModule);
+                moduleSpecification.addSystemDependency(new ModuleDependency(moduleLoader, MODULE_MAPPER_ORM_DEFAULT,
+                        false, true, true, false));
+                ROOT_LOGGER.debugf("deployment %s contains %s annotation, added %s dependency", deploymentUnit.getName(),
+                        ANNOTATION_INDEXED_NAME, MODULE_MAPPER_ORM_DEFAULT);
+            }
+            else {
+                // Hibernate Search 5.x support
+                // TODO Remove this when we switch to EE 10 and remove Hibernate Search 5.x modules
```

Comment:
Should there be a single tracking WFLY jira for the various TODOs?

Reply:
Sure. I created https://issues.redhat.com/browse/WFLY-16362 and referenced it in every TODO.